packages/did/src/DidDetails/LightDidDetails.ts

```diff
@@ -8,245 +8,163 @@
+      `Authentication key encoding "${authKeyTypeEncoding}" does not match any supported key type`
+    )
   }
+  const authenticationKey: NewLightDidAuthenticationKey = {
+    publicKey: decodeAddress(identifier.substring(2), false, ss58Format),
+    type: decodedAuthKeyType,
+  }
+  if (!encodedDetails) {
+    return fromDetails({ authenticationKey })
+  }
+  const { encryptionKey, serviceEndpoints } =
+    decodeAndDeserializeAdditionalLightDidDetails(encodedDetails, version)
+  return fromDetails({
+    authenticationKey,
+    encryptionKey,
+    serviceEndpoints,
+  })
+}
 
-  /**
-   * Migrate a light DID to a full DID, while maintaining the same keys and service endpoints.
-   *
-   * @param submitterAddress The KILT address to bind the DID creation operation to. It is the same address that will have to submit the operation and pay for the deposit.
-   * @param sign The callback to sign the operation.
-   * @param migrationCallback A user-provided callback to handle the packed and ready-to-be-signed extrinsic representing the DID creation operation.
-   * @param upgradeOptions Optional.
-   * @param upgradeOptions.withEncryptionKey When set to true (default) the LightDID's encryption key is added to the on-chain DID.
-   * @param upgradeOptions.withServiceEndpoints When set to true the LightDID's ServiceEndpoints are added to the on-chain DID. This is strictly opt-in as there are more restrictive size limits for on-chain service records.
-   * @returns The migrated [[FullDidDetails]] if the user-provided callback successfully writes the full DID on the chain. It throws an error otherwise.
-   */
-  public async migrate(
-    submitterAddress: IIdentity['address'],
-    sign: SignCallback,
-    migrationCallback: DidMigrationCallback,
-    { withEncryptionKey = true, withServiceEndpoints = false } = {}
-  ): Promise<FullDidDetails> {
-    const api = await BlockchainApiConnection.getConnectionOrConnect()
-    const creationTx = await FullDidCreationBuilder.fromLightDidDetails(
-      api,
-      this,
-      { withEncryptionKey, withServiceEndpoints }
-    ).build(sign, submitterAddress)
-
-    await migrationCallback(creationTx)
-
-    const fullDidDetails = await FullDidDetails.fromChainInfo(
-      getKiltDidFromIdentifier(this.identifier, 'full')
-    )
-    if (!fullDidDetails) {
-      throw new SDKErrors.DidError('Something went wrong during the migration')
-    }
-    return fullDidDetails
+/**
+ * Create a new instance of [[LightDidDetails]] from the provided KILT address.
+ * The resulting DID will only have an authentication key, and no encryption key nor service endpoints.
+ *
+ * @param identifier The KILT address to generate the DID from.
+ * @param keyType One of the [[LightDidSupportedVerificationKeyType]]s to set the type of the authentication key derived from the provided address. It defaults to Sr25519.
+ *
+ * @returns The resulting [[LightDidDetails]].
+ */
+export function fromIdentifier(
```

Comment:
same here

Reply:
This one is only used in a few tests and I wanted to remove it completely

Reply:
Removed in a4b3901d